### PR TITLE
Point to the dedicated Java support doc

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: docs/source/migration/4.0.mdx [hunk 1]
```diff
@@ -203,8 +203,6 @@ Over the years, a lot of support functionality was added alongside `apollo-api`
 
 Moving forward, some artifacts are released separately and have new Maven coordinates, GitHub repositories and package names. Others are deprecated.
 
-For the initial v4 release, all the artifacts are kept with deprecation warnings. They will be removed in the future.
```
Comment (martinbonnin) [inline docs/source/migration/4.0.mdx:206]: Non-incubating artifacts are still around IIRC?

Comment (BoD) [reply docs/source/migration/4.0.mdx:206]: Yes except `apollo-cli`, but maybe it's OK to leave the sentence anyway.

Comment (martinbonnin) [reply docs/source/migration/4.0.mdx:206]: apollo-cli was `apollo-cli-incubating` in main (but maybe not in v3 in which case I can probably bring it back)

Comment (martinbonnin) [reply docs/source/migration/4.0.mdx:206]: Or maybe we just do "most artifacts are kept...."?

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**